### PR TITLE
Update ec2.py

### PR DESCRIPTION
**Pull request by aioue**
add `ec2_vpd_id` metadata to rds instance inventory information

##### SUMMARY

Inventory did not output VPC ID for RDS instances, despite it being available in the existing call data.

This change appends `ec2_vpc_id` to the outputted inventory data for RDS instances.

This is also the first step in allowing RDS instances to be filtered by VPC (https://github.com/ansible/ansible/issues/10193).

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

ec2.py

##### ANSIBLE VERSION

```
ansible 2.3.1.0
  config file = /Users/tom/repos/aws-codecommit/ansible-tower/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 3.6.1 (default, Apr  4 2017, 09:40:21) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.38)]
```

##### ADDITIONAL INFORMATION

Before
```
      "rds_dev_rds_amazonaws_com": {
        "ansible_ssh_host": "rds.mkxms.eu-central-1.rds.amazonaws.com",
       [...]
        "ec2_status": "available",
      },
```

After
```
      "rds_dev_rds_amazonaws_com": {
        "ansible_ssh_host": "rds.mkxms.eu-central-1.rds.amazonaws.com",
       [...]
        "ec2_status": "available",
        "ec2_vpc_id": "vpc-02985f6a"
      },
```

(ec2_vpc_id added) to list data.

### CODE REVIEW

File: contrib/inventory/ec2.py [hunk 1]
```diff
@@ -647,6 +647,8 @@ def get_rds_instances_by_region(self, region):
                         instance.tags = {}
                         for tag in tags:
                             instance.tags[tag['Key']] = tag['Value']
+                        # collect the VPC ID from the DB subnet group
+                        instance.vpc_id = db_instances['DBInstances'][index]['DBSubnetGroup']['VpcId']
```
Comment (willthames) [inline contrib/inventory/ec2.py]: This doesn't seem to do anything - it adds the information but then doesn't use it for grouping or display, as far as I can tell. Am I missing something?

Comment (aioue) [reply contrib/inventory/ec2.py]: Odd - it's displayed when running the inventory for me. See the output in the ticket. Try disabling the cache (setting it to 0)?

Comment (aioue) [reply contrib/inventory/ec2.py]: The output loops over everything assigned a value inside `instance`, so by adding `vpc_id`, it is automatically displayed and does not need specifying again.

**Comment by aioue** [comment]
@willthames 

### Steps to reproduce

- take this latest commit of `ec2.py`

https://github.com/ansible/ansible/blob/365f3dc9d973aa07910c781d7471f27124d2e478/contrib/inventory/ec2.py

- this `ec2.ini` config, with the cache turned off

https://github.com/ansible/ansible/blob/365f3dc9d973aa07910c781d7471f27124d2e478/contrib/inventory/ec2.ini

`cache_max_age = 0`

- run the inventory

#### Output of unmodifed inventory

No VPC ID contained in metadata.

```
"foo_us_west_1_rds_amazonaws_com": {
  "ansible_ssh_host": "foo.us-west-1.rds.amazonaws.com",
  "ec2_DBInstance": "",
  "ec2_OptionGroupMembership": "",
  "ec2_OptionGroupMemberships": "",
  "ec2_OptionGroupName": "default:postgres-9-5",
  "ec2_PubliclyAccessible": "false",
  "ec2_ReadReplicaSourceDBInstanceIdentifier": "foo",
  "ec2_Status": "in-sync",
  "ec2__address": "foo.us-west-1.rds.amazonaws.com",
  "ec2__in_endpoint": false,
  "ec2__port": 5432,
  "ec2_account_id": "12345678",
  "ec2_allocated_storage": 500,
  "ec2_arn": "arn:aws:rds:us-west-1:12345678:db:foo",
  "ec2_auto_minor_version_upgrade": true,
  "ec2_availability_zone": "us-west-1c",
  "ec2_backup_retention_period": 0,
  "ec2_character_set_name": "",
  "ec2_create_time": "2017-07-11T06:55:46.840Z",
  "ec2_engine": "postgres",
  "ec2_engine_version": "9.5.2",
  "ec2_id": "foo",
  "ec2_instance_class": "db.m4.large",
  "ec2_iops": "",
  "ec2_latest_restorable_time": "",
  "ec2_license_model": "postgresql-license",
  "ec2_master_username": "postgres",
  "ec2_multi_az": false,
  "ec2_preferred_backup_window": "04:04-04:34",
  "ec2_preferred_maintenance_window": "fri:04:36-fri:05:06",
  "ec2_status": "available"
}
```

#### Output of inventory with change

VPC ID present in metadata.

```
"foo_us_west_1_rds_amazonaws_com": {
  "ansible_ssh_host": "foo.us-west-1.rds.amazonaws.com",
  "ec2_DBInstance": "",
  "ec2_OptionGroupMembership": "",
  "ec2_OptionGroupMemberships": "",
  "ec2_OptionGroupName": "default:postgres-9-5",
  "ec2_PubliclyAccessible": "false",
  "ec2_ReadReplicaSourceDBInstanceIdentifier": "foo",
  "ec2_Status": "in-sync",
  "ec2__address": "foo.us-west-1.rds.amazonaws.com",
  "ec2__in_endpoint": false,
  "ec2__port": 5432,
  "ec2_account_id": "12345678",
  "ec2_allocated_storage": 500,
  "ec2_arn": "arn:aws:rds:us-west-1:12345678:db:foo",
  "ec2_auto_minor_version_upgrade": true,
  "ec2_availability_zone": "us-west-1c",
  "ec2_backup_retention_period": 0,
  "ec2_character_set_name": "",
  "ec2_create_time": "2017-07-11T06:55:46.840Z",
  "ec2_engine": "postgres",
  "ec2_engine_version": "9.5.2",
  "ec2_id": "foo",
  "ec2_instance_class": "db.m4.large",
  "ec2_iops": "",
  "ec2_latest_restorable_time": "",
  "ec2_license_model": "postgresql-license",
  "ec2_master_username": "postgres",
  "ec2_multi_az": false,
  "ec2_preferred_backup_window": "04:04-04:34",
  "ec2_preferred_maintenance_window": "fri:04:36-fri:05:06",
  "ec2_status": "available",
  "ec2_vpc_id": "vpc-77abcb04"
}
```

**Comment by aioue** [comment]
@willthames were you able to test the inventory with the changes?

**Comment by s-hertel** [comment]
Hi, @aioue. Since adding the inventory plugins in 2.4 we haven't been working on adding features to the contrib scripts. You can add flexible hostvars using the inventory plugin's constructed `compose` option. Please take a look at the [aws_rds inventory plugin](https://docs.ansible.com/ansible/latest/plugins/inventory/aws_rds.html) and see if it meets your needs. If not I'd be happy to review a feature for it.

**Pull request closed without merging**